internal/util/update/resource-merge.go

```diff
@@ -131,6 +135,69 @@ func (u ResourceMergeUpdater) updatedKptfile(updatedPath, originalPath string, o
 	return updatedKf, err
 }
 
+// MergeSubPackages merges the Kptfiles in the nested subdirectories of the
+// root package and also sets the setter definitions in updated to match with
+// locally set values so the the resources are correctly identified and merged
+func MergeSubPackages(localRoot, updatedRoot, originalRoot string) error {
+	localPkgPaths, err := pathutil.DirsWithFile(localRoot, kptfile.KptFileName, true)
+	if err != nil {
+		return err
+	}
+	for _, localPkgPath := range localPkgPaths {
+		// skip the top level file as it should be merged differently
+		if filepath.Clean(localPkgPath) == filepath.Clean(localRoot) {
+			continue
+		}
+
+		relativePkgPath := strings.TrimPrefix(localPkgPath, localRoot)
```

Comment:
Do we know that this is always safe? Paths might have `../` in them that could fail if we just do string manipulation. `filepath.Clean` would probably remove these, but we don't keep the results of the calls to `Clean` further up. I think using `filepath.Clean` combined with `filepath.Rel` would be safer here.

Reply:
@mortent Done.